webapp/channels/src/actions/user_actions.ts

```diff
@@ -316,106 +312,110 @@ export const getGMsForLoading = (state: GlobalState) => {
 };
 
 export async function loadProfilesForGM() {
-    const state = getState();
-    const newPreferences = [];
-    const userIdsInChannels = Selectors.getUserIdsInChannels(state);
-    const currentUserId = Selectors.getCurrentUserId(state);
-    const collapsedThreads = isCollapsedThreadsEnabled(state);
+    return async (doDispatch: DispatchFunc, doGetState: GetStateFunc) => {
```

Comment:
let's remove the do prefix from dispatch and getstate, the reason was that we had [these assignments](https://github.com/mattermost/mattermost-server/pull/23474/files#diff-bb0fe5c9daad208e59d025b02e2058f2943e679c44c8e9c0eadc631f4e304176L42-L43) defined from global state

Reply:
Thanks for the feedback and explanation. I've also done this for all the action creators.

Also, thanks for pointing me to an example of [type casting state](https://github.com/mattermost/mattermost-server/blob/c943ed6859b1be013e4da92b1952c607a35258bc/webapp/channels/src/actions/marketplace.ts#L25-L41). And lastly, thanks for triaging issues to help contributors!